daemon/volumes.go

```diff
@@ -121,7 +120,7 @@ func (daemon *Daemon) registerMountPoints(container *container.Container, hostCo
 			return fmt.Errorf("Duplicate mount point '%s'", bind.Destination)
 		}
 
-		if len(bind.Name) > 0 {
```

Comment:
Volumes having a name is not the same as the mountpoint being specified by name.